examples/autoscaling/main.tf

```diff
@@ -2,17 +2,22 @@ variable "region" {
   default = "us-central1"
 }
 
-provider "google" {
+provider "google-beta" {
```

Comment:
cannot we set here the project and prevent several definitions from needing to incorporate that there? This requires the variable project to be reallocated so its declared before this block

Reply:
for the CLA stuff - the link appears to be inteded to resolve to https://opensource.google.com/docs/cla/#troubleshoot
I cannot say what you need to do in your case but the steps would be:

Ensure corporate CLA has been accepted

Using the wrong email address
One of the most common problems is that the git author email in the commit is not an email address associated with a CLA. The solution is to change the git author email to be an address covered by the CLA. 

######

With regard to the provider a number of the resources created by the module appear to be supported only by the 'google_beta' provider. This is my first time using Terraform with GCP- I use AWS professionally. My understanding is that all providers should be defined at the top level, here that will need to be the beta provider. If you can see a better way please patch and test!

Reply:
In regards to the CLA, I thought you were googler, so maybe able to fix that... But yes, I can change the address, and push -f into my branch so that will change the associated email address to the right one! I will do that in my branch, and you will probably need to rebase.

Regarding the code comment, you can use vars in the provider config if they are declared before. So, despite it may look strange, I'd suggest to move the specific project variable just before this block, so you avoid duplication.. but its completely up to you: no solution looks perfect

Reply:
But news about the CLA... i see my commits are already associated to the mail I used to sign the CLA. So I will need to wait for a Googler to bring extra info.. thanks in any case for trying to help!

Reply:
OK so if you let me know when you have updated your branch, I can do mine

I'm sorry to be dim but I really don't understand. My understanding was that Terraform was declarative and made it's own dependency tree. The order in which resources are declared doesn't matter. TBH I find it pretty poor that variables here are all in 'main.tf' rather than a separate 'terraform.vars'.

Reply:
Yep, Terraform is almost declarative :) . This looks as an exception: see [this](https://www.terraform.io/docs/configuration/providers.html) for more info:
```
The configuration arguments defined by the provider may be assigned using expressions, which can for example allow them to be parameterized by input variables. However, since provider configurations must be evaluated in order to perform any resource type action, provider configurations may refer only to values that are known before the configuration is applied
```
I haven't played though with that... 

 Best!!

Reply:
Yeah, _"This page is about Terraform 0.12 and later. "_. Version 0.11(current) doesn't do those evaluations